### PR TITLE
refactor(backend): throw exception instead of returning null

### LINKED ISSUE #707: Don't omit exceptions thrown from backend fetch
https://github.com/aeternity/superhero-ui/blob/a60c0d4bad5fcf85a22d47753d5032d12e1d5c3d/src/utils/backend.js#L5

### PR DESCRIPTION
**Pull request by davidyuk**
Known exceptions:

```
[Vue warn]: Error in mounted hook (Promise/async): "Error: Request failed with 400"

found in

---> <WordDetail> at src/views/WordDetail.vue
       <App> at src/App.vue
         <Root>
```
when opening details of word with a space in the name (https://github.com/aeternity/tipping-community-backend/issues/368)

---
```
GET https://testnet.superhero.aeternity.art/profile/image/ak_2UGL6qArnbWx9fEkJyszssMkCjpu3VeLFBPA6MXYV3APfDSz1k 404
```
Chrome always report 404 even if they handled, we can't avoid it

---

I'm proposing a way how errors should be handled:
- don't throw an error if it is somehow indicated in ui (red color or a separate modal window)
- if an error indicated somehow, it may be printed to console (to be able to debug it in production)

Later all not caught exceptions will show modal asking the user to report it. So we shouldn't have unhandled exceptions, at the same time we shouldn't handle them in a too general fashion (like I'm removing in this PR `.catch(() => null)`).

How to test this PR:
- try all features
- ensure that all of them works correctly
- report about all exceptions in console (red-colored) that weren't indicated in UI and not mentioned in the list above

fixes #707

### CODE REVIEW

File: src/components/layout/sendTip/SendTip.vue [hunk 1]
```diff
@@ -115,31 +115,33 @@ export default {
...
           title: this.$t('components.layout.SendTip.SuccessHeader'),
           body: this.$t('components.layout.SendTip.SuccessText'),
         });
         EventBus.$emit('reloadData');
-      }).catch((e) => {
-        this.sendingTip = false;
-        if (e.code && e.code === 4) {
+      } catch (error) {
+        if (error.code && error.code === 4) {
           return;
```
Comment (thepiwo) [inline src/components/layout/sendTip/SendTip.vue:134]: why return without anything?

File: src/components/tipRecords/Author.vue [hunk 2]
```diff
@@ -48,8 +48,7 @@ export default {
     },
   },
   async mounted() {
-    const profile = await Backend.getProfile(this.address);
-    this.name = profile ? profile.preferredChainName : null;
+    this.name = (await Backend.getProfile(this.address)).preferredChainName;
```
Comment (thepiwo) [inline src/components/tipRecords/Author.vue:51]: we can get the names from the names list in the store, the stored name for an account is also the preferredChainName, as the backend sorts them that way when they are globally fetched. This reduces the number of requests.

File: src/utils/backend.js [hunk 3]
```diff
@@ -108,7 +88,7 @@ export default class Backend {
     return backendFetch(`tips${query}`);
   };
 
-  static addToken = async (address) => backendFetchNoTimeout('tokenCache/addToken', {
+  static addToken = async (address) => backendFetch('tokenCache/addToken', {
```
Comment (thepiwo) [inline src/utils/backend.js:91]: this request can take long, when it times out the user gets to see a non working version of word-bazaar, as his token is not indexed and cached properly yet

File: src/components/layout/sendTip/SendTip.vue (hunk 1 above)
Comment (davidyuk) [reply src/components/layout/sendTip/SendTip.vue:134]: The author wanted to omit this kind of exceptions. It was introduced in https://github.com/aeternity/superhero-ui/commit/0b5506d5e7e24bef9376b15e425cc2cf6ce7ecfd I can't get what is supposed to fix, proposing to remove

File: src/utils/backend.js (hunk 3 above)
Comment (davidyuk) [reply src/utils/backend.js:91]: actually, timeout in `backendFetch` was removed in 22f0ee47c8801f28c09cc839129f11ad2ad78f42

File: src/components/tipRecords/Author.vue (hunk 2 above)
Comment (davidyuk) [reply src/components/tipRecords/Author.vue:51]: I have created #1134 to address this

File: src/components/layout/sendTip/SendTip.vue (hunk 1 above)
Comment (davidyuk) [reply src/components/layout/sendTip/SendTip.vue:134]: The author wanted to omit this kind of exception. It was introduced in https://github.com/aeternity/superhero-ui/commit/0b5506d5e7e24bef9376b15e425cc2cf6ce7ecfd I can't get what is supposed to fix, proposing to remove

**Review by Liubov-crypto — COMMENTED** [review]
Trending - pass
Search - pass 
Change profile pic- pass, but have to refresh a page to see results
Change back pic  - pass 
Change profile info - pass 
Cookies - pass 
Send Tip on the Feed - pass (with some issue)
Retip on the feed - pass (with some issue)
Post without tipping - pass  
Add a comment - pass 
Tip comment on the feed - pass (with some issue)
Pin/Unpin - pass 
Claim - pass 
Report post - pass 
Voting - pass 
Vote - pass 
Revoke Vote - pass 
Delegate voting power - pass 
Create a new poll - pass 
Search - pass 
Meet - pass
Send a tip to user icon/comment - failed, the same on the product. 
Updates - pass 
Faq- pass 

Change profile pic- pass, but have to refresh a page to see results:
FF1
FF2

Got this error when tried to send a tip on the feed (tip was sent):
tip on the feed error
tip on the F
Tip on the feed 2
tip on the feed 3
after tip

Got this error when tried to retip content on the feed (retip was sent):
error when retip
error on FF retip content
errr retip 2
retip 3

Got this error when tried to tip comment on the feed (tip was sent):
tip comment on the feed
tip comments on the feed 1
tip comment 2

Voting:
voting

I accidentally put a space before a wallet address and got this: 
мще 2

Meet:
Meet

The error sent tip in the Meet (the same on the product):
error send tip in meet

Remain:
Go back button affect the iframe wallet 
Different names in Meet and wallet

File: src/components/layout/sendTip/SendTip.vue (hunk 1 above)
Comment (thepiwo) [reply src/components/layout/sendTip/SendTip.vue]: ok :D good question

**Comment by davidyuk** [comment]
> but have to refresh a page to see results

This one is fixed

**Comment by Liubov-crypto** [comment]
> > but have to refresh a page to see results
> 
> This one is fixed

Cool! Please let me know if there is anything else to check.

**Review by AtanasKrondev — APPROVED** [review]

**Comment by davidyuk** [comment]
@Liubov-crypto I've added a workaround for v1 tips/retips, please recheck errors from the last review.

**Comment by Liubov-crypto** [comment]
> @Liubov-crypto I've added a workaround for v1 tips/retips, please recheck errors from the last review.

Retested, LGTM. 

Trending - pass 
Search - pass 
 Change profile pic- pass 
 Change back pic - pass 
Change profile info - pass 
Cookies - pass 
 Send Tip on the Feed - pass 
 Retip on the feed - pass 
 Post without tipping - pass 
Add a comment - pass 
Tip comment on the feed - pass 
Pin/Unpin - pass 
Claim - pass 
 Report post - pass 
 Voting - pass 
Vote - pass 
 Revoke Vote - pass 
Delegate voting power - pass 
Search - pass 
 Meet - pass 
Send a tip to user icon/comment - pass  
Updates - pass 
Faq- pass

**Review by thepiwo — APPROVED** [review]

**Pull request merged**